lib/src/event_mixin.dart

```diff
@@ -18,10 +19,21 @@ import 'package:nyxx/src/models/gateway/events/webhook.dart';
 import 'package:nyxx/src/models/interaction.dart';
 import 'package:nyxx/src/utils/iterable_extension.dart';
 
+mixin EventRestMixin implements Nyxx {
+  @internal
+  StreamController<Event> onEventController = StreamController.broadcast();
```

Comment:
We should avoid exposing controllers as part of the API. Events should be added to a stream in the same place the stream is created, otherwise the code becomes hard to follow.

Moving the stream to the plugin avoids this problem.